pkg/apis/machine/v1alpha1/types.go

```diff
@@ -992,6 +993,15 @@ type AzureOSDisk struct {
 	CreateOption string                     `json:"createOption,omitempty"`
 }
 
+type AzureDataDisk struct {
+	Name         string                     `json:"name,omitempty"`
+	Lun          int32                      `json:"lun,omitempty"`
+	Caching      string                     `json:"caching,omitempty"`
```

Comment:
Need to check for chaching, but afaik it is only supported for premium storage. 
BTW: The same for os disk. Need to double check.

Reply:
I couldn't find a limitation to premium storage in the [docs](https://docs.microsoft.com/en-us/learn/modules/caching-and-performance-azure-storage-and-disks/3-enable-and-configure-azure-vm-disk-cache-by-using-the-azure-portal), and I was able to configure caching for standard disks. 
![image](https://user-images.githubusercontent.com/20767214/74585891-75789600-4fea-11ea-9b52-44e22a80f283.png)

Reply:
I have also double checked this. Caching is possible also standard storage. Thanks!

Reply:
Could we make caching optional and default to `None`?

Reply:
done